packages/@react-aria/numberfield/docs/useNumberField.mdx

```diff
@@ -0,0 +1,318 @@
+the [useButton](useButton.html) hook, using the props returned by `useNumberField`.
+
+**Note:** Due to [a bug](https://bugs.webkit.org/show_bug.cgi?id=219188) in Safari on macOS, pointer events may not
+be dispatched after a `<button>` element is disabled while the mouse is pressed. This may require the user to click
+twice when incrementing or decrementing the value from the minimum or maximum value. While out of scope for this example,
+you may wish to use a `<div>` element instead of a `<button>` to avoid this issue. See the [useButton docs](useButton.html#custom-element-type)
+for an example of a button with a custom element type.
+
+```tsx example export=true
+import {useNumberFieldState} from '@react-stately/numberfield';
+import {useLocale} from '@react-aria/i18n';
+import {useButton} from '@react-aria/button';
+
+function NumberField(props) {
+  let {locale} = useLocale();
+  let state = useNumberFieldState({...props, locale});
+  let inputRef = React.useRef();
+  let {
+    labelProps,
+    groupProps,
+    inputProps,
+    incrementButtonProps,
+    decrementButtonProps
+  } = useNumberField(props, state, inputRef);
+
+  let {buttonProps: incrementProps} = useButton(incrementButtonProps);
+  let {buttonProps: decrementProps} = useButton(decrementButtonProps);
+
+  return (
+    <div>
+      <label {...labelProps}>{props.label}</label>
+      <div {...groupProps}>
+        <button {...decrementProps}>-</button>
+        <input {...inputProps} ref={inputRef} />
+        <button {...incrementProps}>+</button>
+      </div>
+    </div>
+  );
+}
+
+<NumberField
+  label="Price"
+  defaultValue={6}
+  formatOptions={{
+    style: 'currency',
+    currency: 'USD'
+  }} />
+```
+
+## Usage
+
+The following examples show how to use the `NumberField` component created in the above example.
+
+### Controlled
+
+By default, `NumberField` is uncontrolled. However, when using the `value` prop, it becomes controlled.
+This allows you to store the current value in your own state, and use it elsewhere.
+
+The `onChange` event is triggered whenever the number value updates. This happens when the user types a
+value and blurs the input, or when incrementing or decrementing the value. It does not happen as the user
+types because partial input may not be parseable to a valid number.
+
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState(6);
+
+  return (
+    <>
+      <NumberField
+        label="Controlled value"
+        value={value}
+        onChange={setValue} />
+      <div>Current value prop: {value}</div>
+    </>
+  );
+}
+```
+
+### Decimals
+
+The default formatting style for `NumberField` is decimal, but you can configure various aspects via the `formatOptions`
+prop. All options supported by [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat)
```

Comment:
this is almost true, we don't support engineering notation or scientific, should we say that?
we probably will support those later, as Number('4.874915326E7') will parse, but I don't know how much work is needed in the other numeral systems and locales

Reply:
ah yeah, will clarify